bloomrun.js

```diff
@@ -43,6 +68,33 @@ BloomRun.prototype.add = function (pattern, payload) {
   return this
 }
 
+BloomRun.prototype.remove = function (pattern, payload) {
+  var matches = matchingBuckets(this._buckets, pattern)
+  payload = payload || pattern
+
+  if (matches.length > 0) {
+    for (var i = 0; i < matches.length; i++) {
+      var bucket = matches[i]
+
+      if (removePattern(bucket, pattern, payload)) {
+        removeBucket(this._buckets, bucket)
+
+        var that = this
+
+        Object.keys(pattern).forEach(function (key) {
+          that._properties.delete(key)
+        })
```

Comment:
You can avoid setting `that` and creating a function here, declare them at top scope and pass a `this` parameter to `forEach`. See https://github.com/mcollina/bloomrun/blob/master/lib/genKeys.js for an example (and the docs!)

Allocating functions within loop is particularly bad!